src/lib/flow/sol-flow-parser.c

```diff
@@ -409,8 +423,47 @@ create_fbp_type(
     return 0;
 }
 
+#ifdef JAVASCRIPT
+static int
+create_js_type(
+    struct parse_state *state,
+    struct sol_str_slice name,
+    struct sol_str_slice contents,
+    const struct sol_flow_node_type **type)
+{
+    const struct sol_flow_parser_client *client = state->parser->client;
+    const char *buf, *filename;
+    struct sol_flow_node_type *result;
+    size_t size;
+    int err;
+
+    SOL_NULL_CHECK(client, -ENOSYS);
+    SOL_NULL_CHECK(client->read_file, -ENOSYS);
+
+    filename = sol_arena_strdup_slice(state->arena, contents);
```

Comment:
If `filename` won't be needed outside `create_js_type()`, why not just use `strndupa()`?